### PR TITLE
CR073 Add Fuel Type

### PR DESCRIPTION
**Pull request by tim-rivett**
Add fuel type to match NeTEX into siri_facilities.xsd

### CODE REVIEW

**Comment by hennoit** [comment]
I think you should leave out the _choice_ structure (and keep only _sequence_). In the current form you cannot have both _FuelType_ AND _FuelTypeOther_, but in the comments you say that _FuelTypeOther_ is required when _FuelType_=other!

The alternative is to leave out the _sequence_ (and keep only _choice_). The name _FuelTypeOther_ will already indicate that it is an 'other' value. In this case of course also leave out "minOccurs=0" for _FuelTypeOther_.

The automatic checks complain that there is already a FuelTypeEnumeration elsewhere in the xsd, so the pull request should also include a change to that file.

**Review by hennoit — CHANGES_REQUESTED** [review]
see my comment above

**Comment by haeckerbaer** [comment]
@tim-rivett 
Can you update the PR according to remark 1:
"_I think you should leave out the choice structure (and keep only sequence). In the current form you cannot have both FuelType AND FuelTypeOther, but in the comments you say that FuelTypeOther is required when FuelType=other!_"

**Review by hennoit — APPROVED** [review]
Please also change 'FuelTypeOther' in the annotation of TypeOfVehicleFuelEnumeration to 'VehicleFuelTypeOther'.

**Review by Aurige — APPROVED** [review]

File: xsd/siri_model/siri_facility-v2.0.xsd [hunk 1]
```diff
@@ -721,4 +721,37 @@ Rail transport, Roads and road transport
    </xsd:sequence>
   </xsd:sequence>
  </xsd:group>
+<!-- ===Vehicle FACILITY====================================== -->
+<xsd:complexType name="TypeOfVehicleFuelStructure">
+	<xsd:annotation>
+		<xsd:documentation>Classification of Vehicle Energy type. See Transmodel TypeOfFuel</xsd:documentation>
+	</xsd:annotation>
+	<xsd:sequence>
+		<xsd:element name="FuelType" type="TypeOfVehicleFuelEnumeration">
```
Comment (haeckerbaer) [inline xsd/siri_model/siri_facility-v2.0.xsd:730]: Please change type to the existing "TypeOfFuelEnumeration" of the siri_journey_support schema

File: xsd/siri_model/siri_facility-v2.0.xsd [hunk 2]
```diff
@@ -721,4 +721,37 @@ Rail transport, Roads and road transport
...
+			<xsd:annotation>
+				<xsd:documentation>Values for Vehicle fuel type. +SIRI v2.1</xsd:documentation>
```
Comment (haeckerbaer) [inline xsd/siri_model/siri_facility-v2.0.xsd:732]: Please change the documentation to the text of TypeOfVehicleFuelEnumeration below: "Values for Vehicle fuel type. Use of "other" requires the additional open ended VehicleFuelTypeOther to be filled."

File: xsd/siri_model/siri_facility-v2.0.xsd [hunk 3]
```diff
@@ -721,4 +721,37 @@ Rail transport, Roads and road transport
...
+			</xsd:annotation>
+		</xsd:element>
+		<xsd:element name="VehicleFuelTypeOther" type="TypeOfValueStructure" minOccurs="0">
+			<xsd:annotation>
+				<xsd:documentation>An arbitrary vehicle fuel classification. +SIRI v2.1</xsd:documentation>
+			</xsd:annotation>
+		</xsd:element>
+	</xsd:sequence>	
+</xsd:complexType>
+<xsd:simpleType name="TypeOfVehicleFuelEnumeration">
```
Comment (haeckerbaer) [inline xsd/siri_model/siri_facility-v2.0.xsd:742]: Please add the new value "unknown" to the existing type "TypeOfFuelEnumeration" of the siri_journey_support schema and then remove this new (and now obsolete" simpleType "TypeOfVehicleFuelEnumeration".

**Review by haeckerbaer — CHANGES_REQUESTED** [review]

**Comment by haeckerbaer** [comment]
@Ulf9  @tim-rivett @hennoit @Aurige 
I changed it slightly to align it with CR002 / new formation model from NeTEx. Please have a look.

**Review by haeckerbaer — APPROVED** [review]

**Review by Aurige — APPROVED** [review]

**Review by tim-rivett — COMMENTED** [review]
Looks good, thank you for the updates.

File: xsd/siri_model/siri_facility-v2.0.xsd [hunk 4]
```diff
@@ -721,37 +721,22 @@ Rail transport, Roads and road transport
...
-</xsd:simpleType>
+ <!-- ===Vehicle FACILITY====================================== -->
+ <xsd:complexType name="TypeOfVehicleFuelStructure">
+  <xsd:annotation>
+   <xsd:documentation>Classification of vehicle energy type. See Transmodel TypeOfFuel. +SIRI v2.1</xsd:documentation>
+  </xsd:annotation>
+  <xsd:sequence>
+   <xsd:element name="TypeOfFuel" type="TypeOfFuelEnumeration">
+    <xsd:annotation>
+     <xsd:documentation>Values for vehicle fuel type. Use of 'other' requires the additional open ended VehicleFuelTypeOther to be filled.</xsd:documentation>
```
Comment (hennoit) [inline xsd/siri_model/siri_facility-v2.0.xsd:732]: Please change "VehicleFuelTypeOther" in the annotation to "OtherTypeOfFuel".

**Review by hennoit — CHANGES_REQUESTED** [review]
One small corretion to the annotations because the name of the element it refers to has changed (again).

File: xsd/siri_model/siri_facility-v2.0.xsd (hunk 4 above)
Comment (haeckerbaer) [reply xsd/siri_model/siri_facility-v2.0.xsd]: See commit [57a9e66](https://github.com/SIRI-CEN/SIRI/pull/58/commits/57a9e66b98e77a06a9a15405a27f81466c350dff)

**Pull request merged**